src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -544,7 +544,14 @@ struct Analyzer {
     finder.walk(curr);
 
     for (auto element : finder.elements) {
-      referenced.insert(element);
+      // Avoid repeated work. Note that globals with multiple references to
+      // previous globals can lead to exponential work, so this is important.
+      // (If C refers twice to B, and B refers twice to C, then when we process
+      // C we would, naively, scan B twice and C four times.)
+      auto [_, inserted] = referenced.insert(element);
+      if (!inserted) {
```

Comment:
Alternatively:
```suggestion
      if (!referenced.insert(element).second) {
```

Reply:
I think it is clearer as it is, by giving a name that indicates we are checking for insertion? Otherwise one needs to memorize what ".second" is...